obelisk/client.py

```diff
@@ -74,6 +77,16 @@ def __init__(self, address, public_key=None, log=None, heartbeat_port=9092):
         self.listen_heartbeat(heartbeat_port)
         task.LoopingCall(self.renew_subscriptions).start(120, now=False)
 
+    def _reconnect(self):
+        if self.addresses is not None:
+            try:
+                index = self.addresses.index(self.address)
+                index = (index + 1) % len(self.addresses)
+                self.address = self.addresses[index]
+            except ValueError:
+                pass
+        return ClientBase._reconnect(self)
+
     def listen_heartbeat(self, port):
         def timeout():
             self.connected = False
```

Comment:
We also should change servers if the hearbeat times out. The _reconnect call only times out on individual calls, but we might know before then if the server went down via the heartbeat.